sdk/storage/azure-storage-blob-nio/README.md

```diff
@@ -76,55 +77,174 @@ b. Alternatively, get the Account SAS Token from the Azure Portal.
+latency. Because of this, particular care must be taken when managing concurrency. Race conditions are more likely to 
+manifest, network failures occur more frequently than disk failures, and other such distributed application scenarios 
+must be considered when working with this FileSystem. While the AzureFileSystem will ensure it takes appropriate steps 
+towards robustness and reliability, the application developer must also design around these failure scenarios and have 
+fallback and retry options available.
+
+The view of the FileSystem from within an instance of the JVM will be consistent, but the AzureFileSystem makes no 
+guarantees on behavior or state should other processes operate on the same data. The AzureFileSystem will assume that it 
+has exclusive access to the resources stored in Azure Blob Storage and will behave without regard for potential 
+interfering applications
+
+Finally, this implementation has currently chosen to always read/write directly to/from Azure Storage without a local 
+cache. Our team has determined that with the tradeoffs of complexity, correctness, safety, performance, debuggability, 
+etc. one option is not inherently better than the other and that this choice most directly addresses the current known
+use cases for this project. While this has consequences for every API, of particular note is the limitations on writing
+data. Data may only be written as an entire file (i.e. random IO or appends are not supported), and data is not 
+committed or available to be read until the write stream is closed. 
 
-1. Go to your Storage Account
-2. Select `Access keys` from the menu on the left
-3. Under `key1`/`key2` copy the contents of the `Connection string` field
+## Examples
 
-## Key concepts
+The following sections provide several code snippets covering some of the most common Azure Storage Blob NIO tasks, including:
 
-Blob Storage is designed for:
+- [Create a `FileSystem`](#create-a-filesystem)
+- [Create a directory](#create-a-directory)
+- [Iterate over directory contents](#iterate-over-directory-contents)
+- [Read a file](#read-a-file)
+- [Write to a file](#write-to-a-file)
+- [Copy a file](#copy-a-file)
+- [Delete a file](#delete-a-file)
+- [Read attributes on a file](#read-attributes-on-a-file)
+- [Write attributes to a file](#write-attributes-to-a-file)
 
-- Serving images or documents directly to a browser
-- Storing files for distributed access
-- Streaming video and audio
-- Writing to log files
-- Storing data for backup and restore, disaster recovery, and archiving
-- Storing data for analysis by an on-premises or Azure-hosted service
+### Create a `FileSystem`
 
-## Examples
+Create a `FileSystem` using the [`shared key`](#get-credentials) retrieved above.
+
+Note that you can further configure the file system using constants available in `AzureFileSystem`.
+Please see the docs for `AzureFileSystemProvider` for a full explanation of initializing and configuring a filesystem
+
+```java
```

Comment:
There should be a way to make a file and add all these snippets in it to catch any errors. Let me go find a PR where I did that

Reply:
https://github.com/Azure/azure-sdk-for-java/pull/7751